pkg/serializer/serializer.go

```diff
@@ -152,6 +158,11 @@ func NewSerializer(forwarder forwarder.Forwarder) *Serializer {
 	return s
 }
 
+// AttachOrchestratorForwarder sets a dedicated forwarder for orchestrator metadata payloads
+func (s *Serializer) AttachOrchestratorForwarder(forwarder forwarder.Forwarder) {
+	s.orchestratorForwarder = forwarder
+}
+
```

Comment:
I guess I wanted to avoid modifying the signature of the main function for something that is optional and not used in the majority of the agents, happy to change it if you think there's a more elegant way to do it 👍

Reply:
I like the idea that the constructor init eveything and then it's read only. it avoids any race condition. You init it and then you can use it in any way you want.

Reply:
I'm OK adding it to the main constructor, but just as a safety check it means I'll modify calls to `NewSerializer` in all agents: security, dogstastd, tests, giving an extra `nil` as parameter in a few places, sounds good?

Reply:
done in https://github.com/DataDog/datadog-agent/pull/7403/commits/6abf80d0d69857ef617f3e120e4443dd99ecfbc0 ✅